src/storage-client/src/controller.rs

```diff
@@ -553,6 +555,40 @@ pub trait StorageController: Debug {
     /// Identifiers not present in `policies` retain their existing read policies.
     fn set_read_policy(&mut self, policies: Vec<(GlobalId, ReadPolicy<Self::Timestamp>)>);
 
+    /// Acquires and returns the desired read holds, advancing them to the since
+    /// frontier when necessary.
+    fn acquire_read_holds(
```

Comment:
I was a little confused as to the purpose of these read holds' ephemeral nature; took me a bit to understand that this is essentially trying to create a cross-sub-system txn where we:
1. Install an ephemeral read hold in the storage controller.
2. Allow the compute controller to instead the read hold at the ephemeral read hold's since
3. Release the ephemeral read hold, knowing that the compute controller has gotten what it needs.

Is that right? If so, I'd love to see this motivating use more explicitly suggested in either the comments, module docs, etc. I see that it's mentioned when you `drop` these read holds, but it was unclear to me that I would find the answer  to "why do we need these to be ephemeral?" exactly there and nowhere else. It seems like a fundamental enough tenet of their design that it could be brought up somewhere.

Reply:
clarified here: https://github.com/MaterializeInc/materialize/pull/26700#issuecomment-2089204947